thunder/distributed/transforms/ddp.py

```diff
@@ -202,3 +202,66 @@ def optimize_allreduce_in_ddp_backward(
     )
     utils.check(visit_callable.has_replaced_return, lambda: "")
     return updated_bwd_trace
+
+
+def apply_bucketing_to_grad_allreduce(trace: TraceCtx) -> TraceCtx:
```

Comment:
It looks like this trace can work not only with gradients but any allreduce inputs?
```suggestion
def apply_bucketing_to_allreduce(trace: TraceCtx) -> TraceCtx:
```

Reply:
I still think this is only for gradient all-reduce as it currently (a) checks `use_ddp` attr of `compile_data.fn` and (b) calls the method to batching gradient all-reduce's or to convert the existing ones into in-place ones.